src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -19,9 +22,18 @@ export function ThemeProvider({
   children,
 }: ThemeProviderProps) {
   const {unstableGlobalTheming = false} = useFeatures();
+  const {nested} = useParent();
```

Comment:
Instead of creating a new context, I think we can reuse the existing ThemeContext - it will be `undefined` if there has been no ThemeContext.Provider providing a value (see `src/utilities/theme/context.tsx`). That should remove the need for ParentContext entirely.

(and small pettiness: I like prefixing boolean values with `is` (or `has` of grammar requires it)

written but untested:

```ts
const rawTheme = useContext(ThemeContext); // may return undefined. Perhaps create a useRawTheme() hook?
// I updated to latest TS the other week, see if optional chaining works? :D
const parentSurface = rawTheme && rawTheme.UNSTABLE_colors ? rawTheme.UNSTABLE_colors.surface : undefined;
const isNested = !!rawTheme;
```